providers/sendgrid/src/lib/sendgrid.provider.ts

```diff
@@ -64,7 +65,10 @@ export class SendgridEmailProvider implements IEmailProvider {
 
   private createMailData(options: IEmailOptions) {
     return {
-      from: options.from || this.config.from,
+      from: {
+        email: options.from || this.config.from,
+        name: this.config.senderName,
+      },
```

Comment:
I haven't been able to find in Sendgrid documentation and codebase how the sender's name is being set. Not even in their tests is shown: https://github.com/sendgrid/sendgrid-nodejs/blob/e1baddb7bfcdfa825d4106121e9d50cafecb8678/test/typescript/mail.ts
On the other hand, setting the name of the receiver of the email is possible. Could you clarify where did you find that it was possible to be done so I can review the PR properly? Thank you.

Reply:
Ignore this. I found about it. Seems they didn't apply any test cases but there is a list of fields that are flexible:
https://github.com/sendgrid/sendgrid-nodejs/blob/main/docs/use-cases/flexible-address-fields.md

Reply:
I just followed the sendgrid types :D